CMakeLists.txt

```diff
@@ -679,6 +682,13 @@ foreach (THRUST_EXAMPLE_SOURCE IN LISTS THRUST_EXAMPLES)
     endif ()
   endif ()
 
+  if ("MSVC" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
+    # Some examples use unsafe APIs (e.g. fopen) that MSVC will complain about
+    # unless this is set:
+    set_target_properties(${THRUST_EXAMPLE}
+      PROPERTIES COMPILE_DEFINITIONS "_CRT_SECURE_NO_WARNINGS")
```

Comment:
Is this just for `discrete_voronoi`? Maybe we should just refactor that one, or at least open an issue tracking this.

Reply:
We should also document in the comment which things need this so we know the condition to remove it.

Reply:
Replaced `fopen` with `fstream`.